iFixit/react-commerce
# Upgrade to node 18

closes #1756

This PR addresses and closes issue #1756 by standardizing Node.JS to v18 for all environments (development, preview, and production). The upgrade extends to both the frontend and Strapi.

#### Frontend
We compensated for a change brought in by Node.js 17 that deprioritized ipv4 addresses over ipv6 by introducing the env variable --dns-result-order ipv4first across various commands; see [here](https://nodejs.org/api/dns.html#dnssetdefaultresultorderorder). Node.JS 17 made this [change](https://github.com/nodejs/node/issues/40702#issuecomment-958157082) without introducing a happy-eyeballs implementation, which led to addresses often resolving to the ipv6 loopback address ::1 instead of 127.0.0.1, thus breaking builds. Notably, Node 20 included a [happy-eyeballs implementation](https://github.com/nodejs/node/pull/44731), and we anticipate being able to remove these env variables in the future.

#### Strapi
Our current version of Strapi advises the use of Node 18 as per their documentation [here](https://docs.strapi.io/dev-docs/installation/cli). To reflect this, we have updated the backend's package.json to only permit the use of the advised Node.JS versions ("node": ">=14.x.x <=18.x.x") and we updated the Strapi docker base image.

### QA

1. Visit Vercel preview.
2. Validate that no discernible changes in backend, frontend, or test functionalities are present.

Node.JS 18 is already set as the build image for the react-commerce Vercel project. Please note that react-commerce-prod will also be updated just before this PR is deployed.
So deploy_block 🛑 for the above reason.
